lib/Device.js

```diff
@@ -237,19 +238,26 @@ class Device extends Entity {
     // Returns:                 Nothing
     assign(options) {
         const deviceGroup = new DeviceGroup(options);
-        this.getEntity().
+        let devices = options.deviceIdentifier.map(deviceId => new Device().initByIdentifier(deviceId));
+        this._helper.makeConcurrentOperations(devices, device => device.getEntity()).
             then(() => deviceGroup.getEntity()).
             then(() => {
-                if (this.relatedDeviceGroupId === deviceGroup.id) {
-                    UserInteractor.printInfo(UserInteractor.MESSAGES.DEVICE_ALREADY_ASSIGNED_TO_DG,
-                        this.identifierInfo, deviceGroup.identifierInfo);
+                devices = Utils.getUniqueEntities(devices);
+                if (devices.every(device => device.relatedDeviceGroupId === deviceGroup.id)) {
+                    UserInteractor.printInfo(
+                        UserInteractor.MESSAGES.DEVICE_ALREADY_ASSIGNED_TO_DG,
+                        UserInteractor.getMultipleEntitiesInfo(devices),
+                        deviceGroup.identifierInfo);
                 }
                 else {
+                    devices = devices.filter(device => device.relatedDeviceGroupId !== deviceGroup.id);
+                    const reassignedDevices = devices.filter(device => device.assigned);
                     return UserInteractor.processCancelContinueChoice(
-                        Util.format(UserInteractor.MESSAGES.DEVICE_REASSIGN, this.identifierInfo, deviceGroup.identifierInfo),
+                        Util.format(UserInteractor.MESSAGES.DEVICE_REASSIGN, 
+                            UserInteractor.getMultipleEntitiesInfo(reassignedDevices), deviceGroup.identifierInfo),
                         this.assignToDG.bind(this),
-                        [deviceGroup],
-                        options.confirmed || !this.assigned);
+                        [deviceGroup, devices],
+                        options.confirmed || reassignedDevices.length === 0);
                 }
             }).
             then(() => UserInteractor.printResultWithStatus()).
```

Comment:
I noticed in the README that these operations are not necessarily atomic.  In the case that a single device fails, what is the status code returned by `impt`?  (Hoping that it is failure and not success).

Reply:
The status code is failure in this case.